src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -380,6 +380,11 @@ private ConstantValue GetLazyConstantValue(bool earlyDecodingWellKnownAttributes
 #if REPORT_ALL
                     Console.WriteLine("Thread {0}, Field {1}, StartsCycle {2}", Thread.CurrentThread.ManagedThreadId, this, startsCycle);
 #endif
+                    if (this.state.HasComplete(CompletionPart.Attributes))
+                    {
+                        CheckConstantValue(value, diagnostics);
```

Comment:
>CheckConstantValue(value, diagnostics); [](http://example.com/codeflow?start=24&length=39)

Earlier I already made a comment that performing additional validation doesn't belong in this method. If we want to the validation to be performed by the time we reach this point, the caller should be responsible for that.

Reply:
I wanted this to happen only once - when the value is set. If I did it in caller, the error could be reported multiple times - but maybe that's not a problem?